File: py/search/string/BruteForce.py
#!/usr/bin/env python
# -*- coding:gbk -*-

class BruteForce:
    def search(self, text: str, pattern: str) -> int:
        index, patternIndex = 0, 0
        while index < len(text):
            if text[index] != pattern[patternIndex]:
                index += 1 - patternIndex
                patternIndex = 0
                continue

            if patternIndex == len(pattern) - 1:
                return index - patternIndex
            index += 1
            patternIndex += 1
        return -1

File: cpp/search/string/BruteForce.cpp
#include <iostream>
#include <string>
using namespace std;

class BruteForce {
public:
    int search(string text, string pattern) {
        int index = 0;
        int patternIndex = 0;
        while (index < (int) text.size()) {
            if (text[index] != pattern[patternIndex]) {
                index += 1 - patternIndex;
                patternIndex = 0;
                continue;
            }

            if (patternIndex == pattern.size() - 1) {
                return index - patternIndex;
            }
            ++index;
            ++patternIndex;
        }
        return -1;
    }
};
